bin/run.js

```diff
@@ -111,43 +111,35 @@ if (!argv._.length) {
   return;
 }
 
-// Test Pipeline
-const pool = agentPool(Number(argv.threads), hostType, argv.hostArgs, hostPath,
-                       { timeout: argv.timeout, transpiler });
-const paths = globber(argv._);
-if (!includesDir && !test262Dir) {
-  test262Dir = test262Finder(paths.fileEvents[0]);
-}
-const files = paths.map(pathToTestFile);
-const tests = files.map(compileFile);
-const scenarios = tests.flatMap(scenariosForTest);
-const pairs = Rx.Observable.zip(pool, scenarios);
-const rawResults = pairs.flatMap(pool.runTest).tapOnCompleted(() => pool.destroy());
-const results = rawResults.map(test => {
-  test.result = validator(test);
-  return test;
-});
-const resultEmitter = resultsEmitter(results);
-reporter(resultEmitter, reporterOpts);
-
-function printVersion() {
-  const p = require(Path.resolve(__dirname, "..", "package.json"));
-  console.log(`v${p.version}`);
-}
+globber(argv._)
+  .reduce((accumulator, path) => accumulator.concat(path), [])
+  .toPromise()
```

Comment:
Perhaps worth noting that I think in theory the current implementation allows running tests while files are being enumerated (discovery can take some time). But, I'm not sure this worked in practice.